k-Diff.java

```diff
@@ -0,0 +1,44 @@
+//Time complexity: O(N logN)
+//Space complexity: O(1)
```

Comment:
SC is O(n) because if you use mergesort or quicksort for sorting then SC is O(n)